UKHomeOffice/cerberus-service
# Configure Cerberus UI to communicate with Cerberus workflow-service using internal k8s service

## Description

~~**DO NOT MERGE: The changes in this PR will not work until the changes made to Cerberus workflow-service in [THIS PR](https://github.com/UKHomeOffice/workflow-service/pull/197) are deployed** - hence the WIP label~~

This PR:

- Has tidy up - especially around environment variables used for the Cypress test job and associated secret
- Configures the URL for Cerberus workflow-service to point to the cluster internal Kubernetes service rather than the external / public URL. It is hoped this will fix the 504 errors that have been occurring and impacting end users.

The tidy up was needed to make the job of reconfiguring the Cerberus workflow-service URL manageable!

To do:

Once this change goes in, and Cerberus UI is communicating with Cerberus workflow-service using the cluster internal Kubernetes service, the external / public access to Cerberus workflow-service will no longer be required. At this point the external / public access will be removed. This will affect local development and (possibly) the ability to run Cypress tests locally. The README will need updating with new instructions on how to connect to the Dev instance of Cerberus workflow-service using the port forwarding capabilities of `kubectl`

## To Test

- Ensure that the tests are still operating correctly post changes.
- Ensure that local development is still possible using `kubectl` port forwarding to connect to Cerberus workflow-service. 

## Developer Checklist

\* Required
